rfcs/protocols/asset-transfer/generic.md

```diff
@@ -23,38 +23,83 @@ This protocol uses as its building blocks:
+- `GetAssetPledgeStatus(<pledge-id>, <recipient-network-id>, <recipient-user-id>)`: This returns the [AssetPledge structure](../../formats/assets/transfer.md#representing-an-asset-transfer-pledge) on the ledger (or a serialized form of it) corresponding to the `<pledge-id>`. It must validate the recipient network and user identities within the stored pledge against the passed parameter values before returning the information to the caller. It is meant to be used as a lookup function in a cross-network data sharing query (and hence will be accompanied by proof). The recipient's identity must be cryptographically verifiable (typically a certificate).
+- `GetAssetPledgeDetails(<pledge-id>)`: This returns the [AssetPledge structure](../../formats/assets/transfer.md#representing-an-asset-transfer-pledge) on the ledger (or a serialized form of it) corresponding to the `<pledge-id>`. It's purpose is similar to `GetAssetPledgeStatus` except that it doesn't perform validations. It is meant to be used as a lookup function within the network in which the asset has been pledged.
+- `GetAssetClaimStatus(<pledge-id>, <recipient-user-id>, <pledging-user-id>, <pledging-network-id>, <pledge-expiration-time>)`: This returns the [AssetClaimStatus structure](../../formats/assets/transfer.md#representing-claims-on-pledged-assets) on the ledger (or a serialized form of it) corresponding to a given `<pledge-id>`. It must validate the claimant and pledger (both network and user) information in the claim structure against the passed parameter values, and mark the claim's expiration status (i.e., whether the claim can still be made) by comparing the pledge expiration time with the current time. It is meant to be used as a lookup function in a cross-network data sharing query (and hence will be accompanied by proof). The recipient's identity must be cryptographically verifiable (typically a certificate).
+
+## Prerequisites: Smart Contract Developer Responsibilities
```

Comment:
The content of this section is specific to fabric, and can be moved to `fabric.md`. The developer responsibilites for corda is different than what's stated here. And I think it will depend upon how the protocol is implemented in respective DLT platform.